browser/ai_chat/ai_chat_service_factory.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_chat/ai_chat_service_factory.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "base/no_destructor.h"
+#include "brave/browser/brave_browser_process.h"
+#include "brave/browser/misc_metrics/process_misc_metrics.h"
+#include "brave/browser/skus/skus_service_factory.h"
+#include "brave/components/ai_chat/content/browser/model_service_factory.h"
+#include "brave/components/ai_chat/core/browser/ai_chat_credential_manager.h"
+#include "brave/components/ai_chat/core/browser/ai_chat_service.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#include "chrome/browser/browser_process.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/common/channel_info.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/user_prefs/user_prefs.h"
+#include "components/version_info/channel.h"
+#include "content/public/browser/storage_partition.h"
+
+namespace ai_chat {
+
+// static
+AIChatServiceFactory* AIChatServiceFactory::GetInstance() {
+  static base::NoDestructor<AIChatServiceFactory> instance;
+  return instance.get();
+}
+
+// static
+AIChatService* AIChatServiceFactory::GetForBrowserContext(
+    content::BrowserContext* context) {
+  CHECK(context);
+  return static_cast<AIChatService*>(
+      GetInstance()->GetServiceForBrowserContext(context, true));
+}
+
+AIChatServiceFactory::AIChatServiceFactory()
+    : BrowserContextKeyedServiceFactory(
+          "AIChatService",
+          BrowserContextDependencyManager::GetInstance()) {
+  DependsOn(skus::SkusServiceFactory::GetInstance());
+  DependsOn(ModelServiceFactory::GetInstance());
+}
+
+AIChatServiceFactory::~AIChatServiceFactory() = default;
+
+content::BrowserContext* AIChatServiceFactory::GetBrowserContextToUse(
+    content::BrowserContext* context) const {
+  if (!Profile::FromBrowserContext(context)->IsRegularProfile() ||
```

Comment:
use `ProfileSelections::Builder().WithRegular(ProfileSelection::kOwnInstance)`

Reply:
I just merged another PR which adds https://github.com/brave/brave-core/blob/master/browser/ai_chat/ai_chat_utils.cc#L15-L20:

```
bool IsAllowedForContext(content::BrowserContext* context,
                         bool check_policy /*=true*/) {
  Profile* profile = Profile::FromBrowserContext(context);
  return profile->IsRegularProfile() && ai_chat::features::IsAIChatEnabled() &&
         (!check_policy || IsAIChatEnabled(profile->GetPrefs()));
}
```

That's fine to stay that way? I was going to update here to use this new util, but we'd rather use the `ProfileSelections` path instead?

Reply:
shouldn't this be `.WithRegular(ProfileSelection::kOriginalOnly)` to make sure the service isn't built for incognito? Don't we in-fact want the default behavior of `ProfileKeyedServiceFactory`?
> Example of a service that does not exist in OTR (default behavior)